etna/alerts/templatetags/alert_tags.py

```diff
@@ -0,0 +1,51 @@
+from django import template
+
+from wagtail.core.models import Page
+
+from ..models import Alert
+
+# Alert snippets
+
+register = template.Library()
+
+
+def get_page_alert(page):
+    """
+    Get the alert field from page, if it exists.
+    """
+    alert = None
+    try:
+        alert = page.alert
+    except AttributeError:
+        pass
+
+    if alert and alert.active:
+        return alert
+    else:
+        return None
+
+
+@register.inclusion_tag('alerts/tags/alerts.html', takes_context=False)
+def alerts(page):
+    """
+    Return alerts from current page as well as those cascaded from ancestors.
+    """
+    alerts = []
+
+    # Get alert for current page, if any.
+    alert = get_page_alert(page)
+    if alert:
+        alerts.insert(0, alert)
+
+    # Get current page ancestors.
+    branch = Page.objects.ancestor_of(page).specific()
+    # Iterate backwards and, if page has alert and "cascade" is true,
+    # add to alerts list.
+    for i in reversed(range(branch.count())):
```

Comment:
A more idiomatic way to loop through the pages in reverse order is:

```
for page in Page.objects.ancestor_of(page).specific().reverse():
    …
```

Is there a reason why we need to gather the alerts in reverse order but insert them at the front of the list?

Reply:
This code has been simplified. The list reversal is no longer required to return pages in hierarchical order.